apps/studio/src/features/editing-experience/components/CreateCollectionModal/CreateCollectionModal.tsx

```diff
@@ -0,0 +1,185 @@
+import type { UseDisclosureReturn } from "@chakra-ui/react"
+import type { z } from "zod"
+import { useEffect } from "react"
+import {
+  Box,
+  FormControl,
+  FormHelperText,
+  FormLabel,
+  Icon,
+  Input,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  VStack,
+} from "@chakra-ui/react"
+import {
+  Button,
+  FormErrorMessage,
+  ModalCloseButton,
+  useToast,
+} from "@opengovsg/design-system-react"
+import { BiLink } from "react-icons/bi"
+
+import { useZodForm } from "~/lib/form"
+import { createCollectionSchema } from "~/schemas/collection"
+import {
+  MAX_FOLDER_PERMALINK_LENGTH,
+  MAX_FOLDER_TITLE_LENGTH,
+} from "~/schemas/folder"
+import { trpc } from "~/utils/trpc"
+import { generateResourceUrl } from "../utils"
+
+type CreateCollectionProps = z.infer<typeof createCollectionSchema>
+
+type CreateCollectionModalProps = Pick<
+  UseDisclosureReturn,
+  "isOpen" | "onClose"
+> &
+  Pick<CreateCollectionProps, "siteId">
+
+export const CreateCollectionModal = ({
+  isOpen,
+  onClose,
+  siteId,
+}: CreateCollectionModalProps): JSX.Element => {
+  return (
+    <Modal isOpen={isOpen} onClose={onClose}>
+      <ModalOverlay />
+      <CreateCollectionModalContent
```

Comment:
should we put the key here instead of inside the component so the parent triggers the rerender instead of the child?

```suggestion
      <CreateCollectionModalContent
        key={String(isOpen)}
```